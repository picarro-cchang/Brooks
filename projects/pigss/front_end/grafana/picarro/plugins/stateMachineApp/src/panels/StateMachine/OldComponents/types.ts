// import { any } from "prop-types";

// export interface ButtonInfo {
//   caption: string;
//   className: string;
//   response: string;
// }

// export interface CommandPanelOptions {
//   uistatus: {
//     [key: string]: string;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;
//   layoutSwitch: () => {}
// }

// export interface ModalInfo {
//   show: boolean;
//   html: string;
//   num_buttons: number;
//   buttons: { [key: string]: ButtonInfo };
// }

// export interface Plan {
//   max_steps: number;
//   panel_to_show: number;
//   current_step: number;
//   focus: PlanFocus;
//   last_step: number;
//   steps: { [key: string]: PlanStep };
//   num_plan_files: number;
//   plan_files: { [key: string]: string };
//   plan_filename: string;
//   bank_names: {
//     [key: number]: {
//       name: string;
//       channels: { [key: number]: string };
//     };
//   };
// }

// export interface PlanFocus {
//   row: number;
//   column: number;
// }

// export interface PlanLoadPanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   updatePanel: (x: number) => void;
//   loadPlan: (x: string) => void;
//   deleteFile: (x: string) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
//   cancelLoadPlan: () => void;
// }

// export interface PlanLoadPanelEditOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   getPlanFromFileName: (s: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export interface PlanPanelOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   fileName: string;
//   loadFile: () => void;
//   savePlan: () => void;
//   updatePanel: (x: number) => void;
// }

// export interface PlanPanelLayoutOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   updatePanel: (x: number) => void;            
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export enum PlanPanelTypes {
//   NONE = 0,
//   PLAN = 1,
//   LOAD = 2,
//   EDIT = 3,
//   PREVIEW = 4
// }

// export interface PlanSavePanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   planSaved: (f: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   fileNames: {[key: string]: string}
// }

// export interface BankConfig {
//   chan_mask: number;
//   clean: number;
// }

// export interface PlanStep {
//   banks: { [key: number]: BankConfig };
//   reference: number;
//   duration: number;
// }

// export interface EditPanelOptions {
//   uistatus: {
//     [key: string]: any;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;

// }
