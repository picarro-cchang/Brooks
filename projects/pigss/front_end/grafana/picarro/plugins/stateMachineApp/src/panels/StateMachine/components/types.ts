export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
