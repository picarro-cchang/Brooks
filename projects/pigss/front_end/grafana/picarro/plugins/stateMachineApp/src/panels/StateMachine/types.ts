import { any } from "prop-types";

export interface ButtonInfo {
  caption: string;
  className: string;
  response: string;
}

export interface CommandPanelOptions {
  uistatus: {
    [key: string]: string;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}

export interface ModalInfo {
  show: boolean;
  html: string;
  num_buttons: number;
  buttons: { [key: string]: ButtonInfo };
}

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface PlanLoadPanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface PlanPanelOptions {
  uistatus: { [key: string]: string };
  plan: Plan;
  setFocus: (row: number, column: number) => void;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
  // addChanneltoPlan: (bank: number, channel: number) => void;
  bankAddition: {[key: string]: number}
}

export enum PlanPanelTypes {
  NONE = 0,
  PLAN = 1,
  LOAD = 2,
  SAVE = 3,
  EDIT = 4
}

export interface PlanSavePanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}

export interface EditPanelOptions {
  uistatus: {
    [key: string]: any;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}
