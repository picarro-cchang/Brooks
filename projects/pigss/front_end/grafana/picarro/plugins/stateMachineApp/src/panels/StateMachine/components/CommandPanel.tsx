import React, { Component, PureComponent } from "react";
import { CommandPanelOptions } from "./../types";

class CommandPanel extends PureComponent<CommandPanelOptions> {
  classNameOpt = {
    DISABLED: "",
    READY: "btn-outline-success",
    ACTIVE: "btn-green"
  };

  getDisabled = (element: string): boolean => {
    let disabled = true;
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      disabled = status_dict[element] === "DISABLED";
    }
    return disabled;
  };

  getClassNameOpt = (element: string): string => {
    let classNames = "";
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      classNames = (this.classNameOpt as any)[status_dict[element]];
    }
    return classNames;
  };

  render() {
    return (
      <div className="panel-command">
        <div style={{ width: "100%", marginTop: 20 }}>
          <div
            className="grid-command"
            style={{
              display: "grid",
              gridTemplateColumns: "1fr 1fr 1fr",
              gridGap: 5,
              padding: 20
            }}
          >
            <button
              onClick={e => this.props.ws_sender({ element: "standby" })}
              disabled={this.getDisabled("standby")}
              value="standby"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("standby")
              }
            >
              Standby
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "identify" })}
              disabled={this.getDisabled("identify")}
              value="identify"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("identify")
              }
            >
              Identify Available Channels
            </button>

            <button
              onClick={e => {
                this.props.ws_sender({ element: "plan" });
              }}
              disabled={this.getDisabled("plan")}
              value="plan"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("plan")
              }
            >
              Edit Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "run" })}
              disabled={this.getDisabled("run")}
              value="run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("run")
              }
            >
              Run Channel
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "plan_run" })}
              disabled={this.getDisabled("plan_run")}
              value="plan_run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_run")
              }
            >
              Run Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "plan_loop" })}
              disabled={this.getDisabled("plan_loop")}
              value="plan_loop"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_loop")
              }
            >
              Loop Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "reference" })}
              disabled={this.getDisabled("reference")}
              value="reference"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("reference")
              }
            >
              Reference
            </button>
            <button
              value="edit"
              className={"btn btn-large btn-edit btn-danger"}
              onClick={e => this.props.ws_sender({ element: "edit" })}
              disabled={this.getDisabled("edit")}
            >
              Edit Labels
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default CommandPanel;
