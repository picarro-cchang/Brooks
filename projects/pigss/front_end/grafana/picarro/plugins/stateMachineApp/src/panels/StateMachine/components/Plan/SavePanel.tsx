import React, { Component, ReactText } from "react";
import ReactList from "react-list";
import { PlanSavePanelOptions, Plan } from "../types";
import { PlanService } from "../../api/PlanService";

interface State {
  fileName: string,
  fileNames: {},
  plan: Plan
}
class SavePanel extends Component<PlanSavePanelOptions, State> {
  constructor(props) {
    super(props)
    this.state = {
      fileName: "",
      fileNames: this.props.fileNames,
      plan: this.props.plan
    }
    // this.deleteFile = this.deleteFile.bind(this);
    // this.savePlan = this.savePlan.bind(this);
    this.updateFileName = this.updateFileName.bind(this)
  }

  componentDidMount() {
    const plan = this.props.getStateFromSavedData();
    this.setState({plan: plan})
  }

  updateFileName(e) {
    const plan = {...this.state.plan}
    plan.plan_filename = e.target.value
    this.setState({fileName: e.target.value, plan});
  }

  renderItem = (index: number, key: ReactText) => (
    <div className="container" style={{ paddingTop: "5px" }} key={key}>
      <div className="btn-group d-flex" style={{ marginLeft: "0px" }}>
        <button
          type="button"
          className="btn btn-light w-100 btn-small"
          style={{ color: "black" }}
        >
          {this.state.fileNames[index]}
        </button>
        <button
          type="button"
          className="btn btn-danger btn-small"
          onClick={e =>
            {
              this.props.deleteFile(this.state.fileNames[index]);
            }
          }
        >
          X
        </button>
      </div>
    </div>
  );
  render() {
    let length = Object.keys(this.props.fileNames).length

    const saveButtons = {
      
    }
    
    return (
      <div className="panel-save">
        <h2 style={{ color: "white" }}>Save Plan</h2>
        <div className="panel-save-inner">
          <form>
            <ReactList
              itemRenderer={this.renderItem}
              length={length}
              type={"uniform"}
            />
          </form>
        </div>
        <div className="col-sm-12" style={{ marginTop: "20px" }}>
          <input
            onChange={e => {
              this.updateFileName(e)
              // this.setState({fileName: e.target.value})
            }}
            maxLength={28}
            type="text"
            className="form-control input-large"
            style={{
              backgroundColor: "white",
              borderRadius: 3,
              color: "black",
              height: 35
            }}
            placeholder="Filename (without extension)"
          />
        </div>

        <div className="container" style={{ marginTop: "20px" }}>
          <div className="text-center">
            <button
              id="cancel-save"
              type="button"
              onClick={e =>
                this.props.updatePanel(0)
              }
              className={"btn btn-group-2 btn-cancel"}
            >
              Cancel
            </button>
            <button
              type="button"
              id="save-btn"
              onClick={e => {
                // this.savePlan();
                this.props.setModalInfo(true, `<div>Save file as ${this.state.fileName}?</div>`, 2, {
                  1: {
                    caption: "Save",
                    className: "btn btn-success btn-large",
                    response: {filename: this.state.fileName, plan: this.state.plan}
                  },
                  2: {
                    caption: "Cancel",
                    className: "btn btn-danger btn-large",
                    response: null
                  }
                }, 'save')
                
              }}
              className={"btn btn-group-2 btn-green"}
            >
              OK
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default SavePanel;
