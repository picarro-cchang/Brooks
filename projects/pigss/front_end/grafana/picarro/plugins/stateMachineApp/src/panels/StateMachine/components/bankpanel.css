.bank-btn {
    border-radius: 25%;
    /*border: 4px solid #bbbbbbbd;*/
    border: 3px solid hsla(0, 0%, 100%, 0.34);
    height: 90px;
    width: 90px;
    color: black;
    position: relative;
}
.chn-label {
    position: absolute;
    top: 20px;
    left: -0.37em;
    bottom: 0;
    right: 0;
}
.btn-clean {
    width:100%;
    border: 3px solid hsla(0, 0%, 100%, 0.34);
    color: black;
    border-radius: 10px;
}
.btn-clean-act {
    border: none;
}

.test {
    color: black;
}
.btn-group {
    width: 100%;
    /*height: 35px;*/
    border-radius: 10%;
    text-align: center;
    /*margin-right: 20px;*/
    margin-left: -20px;
    font-size: 17px;
}

.btn-row-1 {
    margin-top: 10px;
    align-items: center;
    display: flex;
    justify-content: center;
}
.btn-row-2 {
    margin-top: 10px;
    align-items: center;
    display: flex;
    justify-content: center;
}
.btn-row-3 {
    margin-top: 10px;
    align-items: center;
    style: flex;
    justify-content: center;
}
.panel-plan {
    height: 640px;
    /*background-color: #888;*/
    /*border: 3px solid #111;*/
    /*border-radius: 7px;*/
    /*background-color: #82828287;*/
    background-color: #cacaca;
    padding: 10px;
    /*border: 2px solid #1111117a;*/
    border-radius: 5px;
}
.panel-plan-inner {
    overflow-x: hidden;
    overflow-y: auto;
    max-height: 420px;
    border-radius: 3px;
}
.plan-input {
    background-color: white;
    border: 1px solid black;
    border-radius: 2px;
    /*margin-right: -20px;*/
    color: black;
    margin-bottom: 5px;
}
.panel-bank {
    /*border: 2px solid #111;*/
    border: 2px solid #1111117a;
    margin-right: 2%;
    border-radius: 10px;
    padding:10px;
    float: left;
    background-color: rgba(136, 136, 136, 0.51);

}
.grid-bank {
    display: grid;
    grid-gap: 10px;
    grid-template-columns: 1fr 1fr;
    padding-bottom: 10px;
}
.btn-green {
    background-color:  rgb(86, 166, 75);
}
.btn-1-3 {
    grid-column-start: 1;
    grid-column-end: 4;
    width: 100%;
}
.reference-btn {
    display: inline-block;
}
.ref-btn {
    width: 400px;
    grid-row-start: 2;
    grid-column-start: 1;
    /*margin-top: -100px;*/
}

.btn-command {
    border-radius: 5px;
    color: black;
    height: 80px;
}
.btn-edit {
    border-radius: 5px;
    color: white;
    height: 55px;
    width: 100px;
    margin-right: 20px;
    background-color: red;
    background-image: none;
}
.btn-cancel {
    background-image: none;
    background-color: red;
}
.edit-panel {
    color: black;
    background: white;
    border: 1px solid black;
    border-radius: 2px;
}
.edit-label {
    margin-left: 30px;
    margin-right: 10px;
    font-size: 15px;
    color: black;
}
.panel-edit {
    height: 550px;
    background-color: #cacaca;
    padding: 10px;
    border: 3px solid #111;
    border-radius: 7px;
}
.btn-edit-panel {
    margin-bottom: 10px;
    width: 95px;
    height: 40px;
    border-radius: 5px;
    text-align: center;
    font-size: 20px;
}
.btn-run {
    width: 100%;
}
.btn-inverse.disabled {
    background-color: #333;
    opacity: .9;
}
.btn-success {
    /*background-image: -webkit-linear-gradient(top,#6fe28c,#23843b);*/
    background-color: green;
}

.btn-success:active, .btn-success:focus {
    /*background-image: -webkit-linear-gradient(top,#6fe28c,#23843b);*/
    background-color: green;

}
.btn-run-active {
    background-color: rgba(98, 100, 104, 0.86);
}
.btn-run-issue {
    background-color: #716e6e;
}
.btn-chann {
     background-color:  rgb(86, 166, 75);
 }
.col-bank-name {
    /*max-width: 45%;*/
    /*min-width: 20%;*/
    text-align:left;
    padding-left: 5px;
    padding-right: 5px;
}
.input-small {
    width: 70px;
    margin-left: -10px;
}

.radio-btn {
    display: block;
    position: relative;
    /*padding-left: 35px;*/
    margin-bottom: 12px;
    /*margin-left: 20px;*/
    cursor: pointer;
    font-size: 22px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}
.radio-btn input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
}

.checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
    border-radius: 50%;
}

.radio-btn:hover input ~ .checkmark {
    background-color: #565657;
}

.radio-btn input:checked ~ .checkmark {
    background-color: rgb(31, 96, 196);
}


.checkmark:after {
    content: "";
    position: absolute;
    display: none;
}

.radio-btn input:checked ~ .checkmark:after {
    display: block;
}

.radio-btn .checkmark:after {
    top: 9px;
    left: 9px;
    width: 8px;
    height: 8px;
    border-radius: 50%;
    background: white;
}
.button-edit {
    justify-content: center;
}

.cancel:before {
    content: '\2716';
}

.cancel {
    position: absolute;
    top: 0;
    right: 10%;
    cursor: pointer;
    color: black;
    font-size: 25px;
    display: inline-block;
}

