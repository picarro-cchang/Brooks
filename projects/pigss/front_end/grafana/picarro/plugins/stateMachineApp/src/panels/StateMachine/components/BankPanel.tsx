import React, { PureComponent } from "react";
import "./bankpanel.css";

export interface BankPanelOptions {
  bank: number;
  uistatus: {
    bank?: { [bankNum: string]: string };
    clean?: { [bankNum: string]: string };
    channel?: { [bankNum: string]: { [channelNum: string]: string } };
  };
  ws_sender: (o: object) => void;
  plan: {
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
  };
}

class BankPanel extends PureComponent<BankPanelOptions> {
  bankStyleOpt = {
    READY: { color: "#424242", backgroundColor: "#e4e4e4" },
    ACTIVE: { color: "#fff", backgroundColor: "#56a64b" },
    CLEAN: { color: "#fff", backgroundColor: "#4BBEE3" },
    REFERENCE: { color: "#440", backgroundColor: "#e0b400" }
  };
  cleanClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    ACTIVE: "btn-light",
    CLEAN: "btn-secondary btn-clean-act"
  };
  channelClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    AVAILABLE: "btn-ready",
    ACTIVE: "btn-green",
    CLEAN: "btn-light",
    REFERENCE: "btn-warning"
  };

  render() {
    let bankStyle = {};
    let cleanClassNames = "";
    let cleanDisabled = true;
    let getChannelClassNames = (_: number) => "";
    let getChannelDisabled = (_: number) => true;
    let test = {};

    if ("bank" in (this.props.uistatus as any)) {
      const bankStatus: string = this.props.uistatus.bank && (this.props.uistatus.bank as any)[
        this.props.bank
      ];
      const channelStatus: { [key: number]: string } = this.props.uistatus.channel && (this.props.uistatus
        .channel as any)[this.props.bank];
      const cleanStatus: string = this.props.uistatus.clean && (this.props.uistatus.clean as any)[
        this.props.bank
      ];
      bankStyle = (this.bankStyleOpt as any)[bankStatus];
      cleanClassNames = (this.cleanClassNameOpt as any)[cleanStatus];
      cleanDisabled = cleanStatus !== "READY";
      getChannelClassNames = chan => {
        return this.channelClassNameOpt !== undefined && 
                channelStatus !== undefined && 
                (this.channelClassNameOpt as any)[(channelStatus as any)[chan]];
      }
      getChannelDisabled = chan => { return channelStatus !== undefined && (channelStatus as any)[chan] !== "READY" };
      test = channelStatus;
    }

    const channelButtons = [];
    for (let i = 1; i <= 8; i++) {
      channelButtons.push(
        getChannelDisabled(i) ? (
          <button
            key={i}
            id={"channel-" + i}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
            style={{ color: "black" }}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
                {this.props.plan.bank_names[this.props.bank].channels[i]}
              </u>
            </p>
            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
              {test && test[i]}{" "}
            </p>
          </button>
        ) : (
          <button
            onClick={e => {
              this.props.ws_sender({
                element: "channel",
                bank: this.props.bank,
                channel: i
              });
            }}
            id={"channel-" + i}
            disabled={getChannelDisabled(i)}
            key={i}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
                {this.props.plan.bank_names[this.props.bank].channels[i]}{" "}
              </u>
            </p>

            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
              {test && test[i]}
            </p>
          </button>
        )
      );
    }

    const cleanButton = cleanDisabled ? (
      <div className={"btn btn-large btn-clean " + cleanClassNames}>
        {"Clean"}
      </div>
    ) : (
      <button
        id="clean"
        onClick={e =>
          this.props.ws_sender({ element: "clean", bank: this.props.bank })
        }
        className={"btn btn-large btn-clean " + cleanClassNames}
      >
        {"Clean"}
      </button>
    );
    const value: string = this.props.plan.bank_names[this.props.bank].name;
    return (
      <div>
        <div className="panel-bank" style={bankStyle}>
          <div style={{ width: "100%" }}>
            <h2 style={{ color: "#424242", marginBottom: "10px" }}>{value}</h2>
            <div className="grid-bank">{channelButtons}</div>
            {cleanButton}
          </div>
        </div>
      </div>
    );
  }
}
export default BankPanel;
