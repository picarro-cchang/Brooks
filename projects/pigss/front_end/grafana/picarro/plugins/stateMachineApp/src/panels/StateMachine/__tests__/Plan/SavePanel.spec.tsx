import React from "react";
import { shallow, mount } from "enzyme";
import "jest-styled-components";
import WS from "jest-websocket-mock";
import "jest-fetch-mock";
import Modal from "react-responsive-modal";
import SavePanel from "../../components/Plan/SavePanel";
import { PlanSavePanelOptions, Plan } from "../../components/types";
import mockPlanSavePanel from "../../api/__mocks__/mockSavePanelData.json";

const mockClick = jest.fn(element => {
  return element;
});

const mockUpdatePanel = jest.fn();
const mockPlanSaved = jest.fn()
const mockDeleteFile = jest.fn();
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;
const mockUpdateFileName = jest.fn();
const defaultProps: PlanSavePanelOptions = {
  plan: mockPlanSavePanel,
  ws_sender: mockClick,
  isChanged: false,
  updateFileName: mockUpdateFileName,
  planSaved: mockPlanSaved,
  updatePanel: mockUpdatePanel,
  fileNames: {
    1: "Test"
  },
  deleteFile: mockDeleteFile
};

describe("<PlanSavePanel />", () => {
  const wrapper = shallow(<SavePanel {...defaultProps} />);
  const instance = wrapper.instance() as SavePanel;
  const server = new WS(socketURL);
  const client = new WebSocket(socketURL);

  it("Match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders Items", () => {
    instance.renderItem = jest.fn();
    instance.renderItem(0, 0);
    expect(instance.renderItem).toHaveBeenCalled();

    const fileList = mount(<SavePanel {...defaultProps} />).find(
      "ReactList"
    );
    expect(fileList).toMatchSnapshot();
    expect(
      fileList
        .find("button")
        .at(0)
        .text()
    ).toEqual("Test");
  });

  it("Delete", () => {
    // const spy = jest.spyOn(SavePanel.prototype, 'deleteFile')
    const shal = mount(<SavePanel {...defaultProps}/>)
    const deleteFileName = shal.find("ReactList").find("button").at(1);    
    deleteFileName.simulate("click");
    expect(mockDeleteFile).toHaveBeenCalled();
  });

  it("onChange file name input", () => {
    const event = { target: { name: "input", value: "plan1" } };
    const fileInput = wrapper.find("input");
    fileInput.simulate("change", event);
    expect(wrapper.state("fileName")).toBe("plan1")
  });

  it("Cancel", () => {
    const cancel = wrapper.find("button").at(0);
    cancel.simulate("click");
    expect(mockUpdatePanel).toHaveBeenCalled();
  });

  it("Ok", () => {
    const ok = wrapper.find("button").at(1);
    ok.simulate("click");
    // expect(mockUpdateFileName).toHaveBeenCalled();
    expect(mockPlanSaved).toHaveBeenCalled();
  });
});
