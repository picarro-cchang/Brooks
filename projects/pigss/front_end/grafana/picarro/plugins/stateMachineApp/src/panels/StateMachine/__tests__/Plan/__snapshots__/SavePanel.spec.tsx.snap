// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanSavePanel /> Match Snapshot 1`] = `
<div
  className="panel-save"
>
  <h2
    style={
      Object {
        "color": "white",
      }
    }
  >
    Save Plan
  </h2>
  <div
    className="panel-save-inner"
  >
    <form>
      <ReactList
        axis="y"
        itemRenderer={[Function]}
        itemsRenderer={[Function]}
        length={1}
        minSize={1}
        pageSize={10}
        scrollParentGetter={[Function]}
        scrollParentViewportSizeGetter={[Function]}
        threshold={100}
        type="uniform"
        useStaticSize={false}
        useTranslate3d={false}
      />
    </form>
  </div>
  <div
    className="col-sm-12"
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <input
      className="form-control input-large"
      maxLength={28}
      onChange={[Function]}
      placeholder="Filename (without extension)"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 3,
          "color": "black",
          "height": 35,
        }
      }
      type="text"
    />
  </div>
  <div
    className="container"
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <div
      className="text-center"
    >
      <button
        className="btn btn-group-2 btn-cancel"
        id="cancel-save"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="btn btn-group-2 btn-green"
        id="save-btn"
        onClick={[Function]}
        type="button"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<PlanSavePanel /> Renders Items 1`] = `
<ReactList
  axis="y"
  itemRenderer={[Function]}
  itemsRenderer={[Function]}
  length={1}
  minSize={1}
  pageSize={10}
  scrollParentGetter={[Function]}
  scrollParentViewportSizeGetter={[Function]}
  threshold={100}
  type="uniform"
  useStaticSize={false}
  useTranslate3d={false}
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitTransform": "translate(0px, 0px)",
          "msTransform": "translate(0px, 0px)",
          "transform": "translate(0px, 0px)",
        }
      }
    >
      <div>
        <div
          className="container"
          key="0"
          style={
            Object {
              "paddingTop": "5px",
            }
          }
        >
          <div
            className="btn-group d-flex"
            style={
              Object {
                "marginLeft": "0px",
              }
            }
          >
            <button
              className="btn btn-light w-100 btn-small"
              style={
                Object {
                  "color": "black",
                }
              }
              type="button"
            >
              Test
            </button>
            <button
              className="btn btn-danger btn-small"
              onClick={[Function]}
              type="button"
            >
              X
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</ReactList>
`;
